shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -103,68 +70,281 @@ async fn topology_task_process(
+                                {
+                                    *value
+                                } else {
+                                    return Err(anyhow!("port not an integer"));
+                                }
+                            } else {
+                                9042
+                            };
+
+                            Ok(CassandraNode {
+                                address: SocketAddr::new(ip, port.try_into()?),
+                                rack,
+                                _tokens: tokens,
+                                outbound: None,
+                            })
+                        })
+                        .collect()
+                }
+                operation => Err(anyhow!(
+                    "system.peers returned unexpected cassandra operation: {:?}",
+                    operation
+                )),
+            }
+        } else {
+            Err(anyhow!(
+                "Failed to parse system.local response {:?}",
+                response
+            ))
+        }
+    }
+}
+
+mod system_peers {
+    use super::*;
+
+    pub async fn query(
+        connection: &CassandraConnection,
+        data_center: &str,
+    ) -> Result<Vec<CassandraNode>> {
+        let (tx, rx) = oneshot::channel();
+        connection.send(
+            Message::from_frame(Frame::Cassandra(CassandraFrame {
+                version: Version::V4,
+                stream_id: 0,
+                tracing_id: None,
+                warnings: vec![],
+                operation: CassandraOperation::Query {
+                    query: Box::new(parse_statement_single(
+                        "SELECT native_port, native_address, rack, tokens, data_center FROM system.peers_v2",
+                    )),
+                    params: Box::new(QueryParams::default()),
+                },
+            })),
+            tx,
+        )?;
+
+        let mut response = rx.await?.response?;
+
+        if is_unconfigured_peers_v2_table_response(&mut response) {
+            let (tx, rx) = oneshot::channel();
+            connection.send(
+                Message::from_frame(Frame::Cassandra(CassandraFrame {
+                    version: Version::V4,
+                    stream_id: 0,
+                    tracing_id: None,
+                    warnings: vec![],
+                    operation: CassandraOperation::Query {
+                        query: Box::new(parse_statement_single(
+                            "SELECT peer, rack, tokens, data_center FROM system.peers",
+                        )),
+                        params: Box::new(QueryParams::default()),
+                    },
+                })),
+                tx,
+            )?;
+            response = rx.await?.response?;
+        }
+
+        into_nodes(response, data_center)
+    }
+
+    fn is_unconfigured_peers_v2_table_response(message: &mut Message) -> bool {
```

Comment:
How about: `is_peers_v2_does_not_exist_error`.
That way you dont need to know that "unconfigured" is what cassandra says when a table does not exist.